horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -982,4 +980,150 @@ public QueryResult queryDataSet(Integer datasetId, String jsonpath, boolean arra
+               if (transformers.isEmpty()) {
+                  new DataSet(run.start, run.stop, "",
+                        test.id, run.data, run, 0, run.owner, run.access)
+                        .persist();
+               } else {
+                  for (Transformer t : transformers){
+                     StringBuilder array = new StringBuilder("[");
+                     boolean atStart = true;
+                     for (int nodePos = 0; nodePos <= nodeSize; nodePos += 1 ) {
+                        if (!atStart) {
+                           array.append(",");
+                           atStart = false;
+                        }
+                        JsonNode n = nodesToProcess.get(nodePos);
+                        if (!t.extractors.isEmpty()) {
+                           for ( NamedJsonPath e : t.extractors ){
+                              QueryResult result = queryData(run.id, e.jsonpath, s.uri, true);
+                              String value = result.value;
+                              if (null != value && !"[]".equals(value)) {
+                                 JsonNode nodes = Util.toJsonNode(result.value);
+                                 for (JsonNode node : nodes){
+                                    if (!atStart) {
+                                       array.append(",");
+                                    }
+                                    array.append(node.asText());
+                                 }
+                                 matched = true;
+                              }
+                           }
+                        }
+                     }
+                     array.append("]");
+
+                     DataSet ds = null;
+                     try {
+                        if (t.function == null || t.function.isBlank()) {
+                           ds = new DataSet(run.start, run.stop
+                              , "", test.id
+                              , Util.toJsonNode(array.toString()), run
+                              , 0, run.owner, run.access);
+                        } else {
+                           try {
+                              StringBuilder code = new StringBuilder();
+                              code.append("const __obj = { __obj == "+array.toString()+" }").append(";\n");
+                              code.append("const __func = ").append(t.function).append(";\n");
+                              code.append("__func(__obj);\n");
+                              String result = execute (test, run, code.toString(), "transformation").trim();
+                              ds = new DataSet(run.start, run.stop
+                                 , "", test.id
+                                 , Util.toJsonNode(result), run
+                                 , 0, run.owner, run.access);
+                           } finally {
+                              ds = new DataSet(run.start, run.stop
```

Comment:
I don't think you should create another DataSet in `finally`

Reply:
ok